elements/rh-dialog/rh-dialog.css

```diff
@@ -74,121 +72,133 @@ section {
 }
 
 [part='content'] {
-  overflow-y: auto;
-  overscroll-behavior: contain;
-  max-height: var(--_box-max-height, calc(100vh - var(--rh-space-3xl, 48px)));
-  box-sizing: border-box;
-  border-radius: var(--rh-border-radius-default, 3px);
+  padding: var(--rh-space-xl, 24px);
 }
 
-[part='content'] ::slotted([slot='header']) {
-  margin-top: 0 !important;
+[part='header'] {
+  background-color: var(--rh-color-surface);
+  padding-block-end: var(--rh-space-sm, 6px);
+  position: sticky;
+  top: calc(-1 * var(--rh-space-xl, 24px));
 }
 
-header {
-  position: sticky;
-  top: 0;
-  background-color: var(--rh-color-surface-lightest, #ffffff);
+[part='header'][hidden] + [part='body'] {
+  max-width: calc(100% - var(--rh-space-xl, 24px));
 }
 
-header ::slotted(:is(h1,h2,h3,h4,h5,h6)[slot='header']) {
+@media (min-width: 1200px) {
+  [part='content'] {
+    padding: var(--rh-space-2xl, 32px);
+  }
+
+  [part='header'] {
+    top: calc(-1 * var(--rh-space-2xl, 32px));
+  }
+
+  [part='header'][hidden] + [part='body'] {
+    max-width: calc(100% - var(--rh-space-2xl, 32px));
+  }
+}
+
+[part='content'] ::slotted([slot='header']) {
+  margin-top: 0 !important;
+}
+
+[part='header'] ::slotted(:is(h1,h2,h3,h4,h5,h6)[slot='header']) {
+  font-family: var(--rh-font-family-heading, RedHatDisplay, 'Red Hat Display', 'Noto Sans Arabic', 'Noto Sans Hebrew', 'Noto Sans JP', 'Noto Sans KR', 'Noto Sans Malayalam', 'Noto Sans SC', 'Noto Sans TC', 'Noto Sans Thai', Helvetica, Arial, sans-serif);
   font-size: var(--rh-font-size-heading-sm, 1.5rem);
   font-weight: var(--rh-font-weight-body-text-regular, 400);
-  font-family: 'Red Hat Display', RedHatDisplay, Overpass, Helvetica, sans-serif;
 }
 
 [part='close-button'] {
-  color: var(--rh-color-icon-subtle, #707070);
-  background-color: transparent;
-  border: none;
-  margin: 0;
-  padding: 0;
-  text-align: left;
-  position: absolute;
+  color: var(--rh-color-icon-secondary);
   cursor: pointer;
-  line-height: 24px;
-  padding-block: 0.375rem;
-  padding-inline: var(--rh-space-lg, 16px);
-  top: 0;
-  right: calc(var(--rh-space-xl, 24px) / -3);
+  display: block;
+  inset-block-start: var(--rh-length-xs, 4px);
+  inset-inline-end: var(--rh-length-xs, 4px);
+  position: absolute;
+  z-index: 500;
 }
 
-[part='close-button'] > svg {
-  font-size: 16px;
-  width: var(--rh-space-lg, 16px);
-  aspect-ratio: 1/1;
+[part='close-button']::part(button) {
+  padding: var(--rh-space-xl, 24px);
 }
 
-[part='close-button']:is(:hover, :focus-within, :focus-visible) svg:is(svg, :hover) {
-  fill: var(--rh-color-icon-secondary-on-light, #151515);
+@media (min-width: 1200px) {
+  [part='close-button']::part(button) {
+    padding: var(--rh-space-2xl, 32px);
```

Comment:
should we use a margin here on the rh-button instead of padding the shadow part?